app/cdap/components/PostRunActions/WizardNav/WizardConfigureConfirmStep/index.tsx

```diff
@@ -0,0 +1,110 @@
+import { getCurrentNamespace } from 'services/NamespaceStore';
+import styled from 'styled-components';
+import uuidV4 from 'uuid/v4';
+
+export interface IWizardConfigureConfirmStepProps {
+  state: any;
+  disabled: boolean;
+  dispatch: (action: any) => void;
+  addPostAction: (action: any) => void;
+  mode: string;
+  setShowValidateButton: (show: boolean) => void;
+  setAction: (action: any) => void;
+  propertyErrors: object;
+  onNextButtonClick: (action: any, toggleNextValidate: any) => void;
+}
+
+export const WizardConfigureConfirmStep = (props: IWizardConfigureConfirmStepProps) => {
+  useEffect(() => {
+    props.setShowValidateButton(true);
+  }, []);
+  const [values, setValues] = useState(props.state.values);
+  const [isValidating, setIsValidating] = useState(false);
+
+  useEffect(() => {
+    const action = { ...props.state.selectedAction };
+    action.plugin.properties = values;
+    // below might not be necessary
+    action.properties = values;
+    props.setAction(action);
+  }, [values]);
+
+  const getConfirmedAction = () => {
+    const confirmedAction = props.state.selectedAction;
+    const newValuesPlugin = { ...confirmedAction.plugin };
+    newValuesPlugin.properties = values;
+    return {
+      name: confirmedAction.name,
+      id: confirmedAction.id || confirmedAction.name + uuidV4(),
+      plugin: newValuesPlugin,
+      description: confirmedAction.description,
+    };
+  };
+
+  return (
+    <React.Fragment>
+      <div className="confirm-step-content">
+        <ConfigurationGroup
+          values={values}
+          pluginProperties={props.state.selectedAction._backendProperties}
+          widgetJson={props.state.widgetJson}
+          disabled={props.disabled}
+          onChange={setValues}
+          errors={props.propertyErrors}
+        />
+      </div>
+      <div className="confirm-step-wizard-footer">
+        <div className="pull-right">
+          {props.state.currentStage > 2 || props.mode !== 'edit' ? (
+            <div
+              className="btn btn-grey-cancel"
+              onClick={() => {
+                props.dispatch({
+                  type: 'back',
+                });
+              }}
+            >
+              Back
+            </div>
+          ) : null}
+
+          {props.state.currentStage === 2 && (
+            <div
+              className="btn btn-blue confirm-step-next-btn"
+              data-cy="next-btn"
+              onClick={() => {
+                setIsValidating(true);
+                props.onNextButtonClick(getConfirmedAction(), setIsValidating);
+              }}
+            >
+              {isValidating ? (
+                <span>
+                  <span className="fa fa-spinner fa-spin"></span>
+                </span>
+              ) : (
+                <span>Next</span>
+              )}
+            </div>
+          )}
+
+          {props.state.currentStage === 3 && (
+            <div
```

Comment:
in general we want to avoid using divs as button - if they _absolutely_ have to be divs for some reason look up the extra stuff to make them accessible and add that.